src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php

```diff
@@ -99,7 +99,12 @@ public function setUser($user)
             throw new \InvalidArgumentException('$user must be an instanceof UserInterface, an object implementing a __toString method, or a primitive string.');
         }
 
-        if (null === $this->user) {
+        // @deprecated since Symfony 5.4, remove the whole block if/elseif/else block in 6.0
+        if (2 === \func_num_args() && !func_get_arg(1)) {
```

Comment:
we should use `<` comparisons everywhere

Reply:
should be better now

Reply:
not quite here :)

Reply:
🤦 fixed!